src/Mapping/ClassMetadata.php

```diff
@@ -2465,12 +2465,16 @@ public function getAssociationMappedByTargetField(string $assocName): string
     }
 
     /**
-     * @return string|null null if the input value is null
-     * @psalm-return class-string|null
+     * @param C $className
+     *
+     * @return string|null null if and only if the input value is null
+     * @psalm-return (C is string ? string : null)
+     *
+     * @template C of string|null
      */
     public function fullyQualifiedClassName(string|null $className): string|null
```

Comment:
Does it make sense to call this method with `null`? Should we rather deprecate that case?

Reply:
I think it would. Right now, it is only useful for a call from `setCustomRepositoryClass`. Let me push an extra commit that adds the deprecation.